injector/network_disruption.go

```diff
@@ -66,6 +69,30 @@ type tcServiceFilter struct {
 	priority uint32 // one priority per tc filters applied, the priority is the same for all interfaces
 }
 
+// tcFilter describes a tc filter
+type tcFilter struct {
+	ip       *net.IPNet
+	priority uint32 // one priority per tc filters applied, the priority is the same for all interfaces
+}
+
+func (t tcFilter) String() string {
+	ip := ""
+	if t.ip != nil {
+		ip = t.ip.String()
+	}
+
+	return fmt.Sprintf("ip=%s; priority=%s", ip, strconv.FormatUint(uint64(t.priority), 10))
+}
+
+func FiltersToStrings(t []tcFilter) []string {
```

Comment:
I don't really understand the reason, but even though `tcFilter` has a String(), and I can print one in a log just fine, trying to print a slice of them just gives me `{[], [], []}`, thus this function

Reply:
It seems due to the fact behing the scene zap uses json marshalling of the struct:
* https://go.dev/play/p/4aOAhZSh7Va

Hence you should have exported fields at least to have something displayed when using it.

You could still have a dedicated type `[]tcFilter` and associate a custom marshaller to reuse the `Stringer` method of the underlying private type if you prefer.